test/unit/org/apache/cassandra/inject/ActionBuilder.java

```diff
@@ -0,0 +1,294 @@
+    }
+
+    public Class<?> getHelperClass()
+    {
+        return helperClass;
+    }
+
+    public ConditionsBuilder conditions()
+    {
+        return conditionsBuilder;
+    }
+
+    public BindingsBuilder bindings()
+    {
+        return bindingsBuilder;
+    }
+
+    public ActionsBuilder actions()
+    {
+        return actionsBuilder;
+    }
+
+    /**
+     * When an invoke point is set on a method in an interface or in some base class and is not overridden in the
+     * class where we want the injection to be executed, we need to add some additional condition. This method
+     * configures
+     * the action to be invoked only in the instance of the specified class.
+     */
+    public ActionBuilder onlyInClass(Class<?> targetClass)
+    {
+        return onlyInClass(targetClass.getName());
+    }
+
+    /** @see #onlyInClass(Class) */
+    public ActionBuilder onlyInClass(String targetClassName)
+    {
+        bindings().addBinding("thisClassName", "String", expr(THIS).method("getClass").args().method("getName").args().toString());
+        conditions().when(expr("thisClassName").method("equals").args(quote(targetClassName)));
+        return this;
+    }
+
+    /**
+     * the action to be invoked only if it's called from given callerMethodName
+     */
+    public ActionBuilder callerEquals(String callerMethodName)
+    {
+        conditions().when(expr("callerEquals").args(quote(callerMethodName)));
+        return this;
+    }
+
+    private enum LogicOp
+    {
+        AND, OR, NOT
+    }
+
+    public class ConditionsBuilder extends Builder
+    {
+        private final LinkedList<Object> elements = new LinkedList<>();
+
+        public ConditionsBuilder clear()
+        {
+            elements.clear();
+            return this;
+        }
+
+        public ConditionsBuilder when(Object expression)
+        {
+            if (!elements.isEmpty() && !(elements.getLast() instanceof LogicOp))
+            {
+                elements.add(LogicOp.AND);
+            }
+            elements.add(expression);
+            return this;
+        }
+
+        public ConditionsBuilder not()
+        {
+            if (!(elements.getLast() instanceof LogicOp))
+            {
+                elements.add(LogicOp.AND);
+            }
+            elements.add(LogicOp.NOT);
+            return this;
+        }
+
+        public ConditionsBuilder and()
+        {
+            Preconditions.checkState(!(elements.getLast() instanceof LogicOp));
+            elements.add(LogicOp.AND);
+            return this;
+        }
+
+        public ConditionsBuilder or()
+        {
+            Preconditions.checkState(!(elements.getLast() instanceof LogicOp));
+            elements.add(LogicOp.OR);
+            return this;
+        }
+
+        public ConditionsBuilder trueLiteral()
+        {
+            return when("TRUE");
+        }
+
+        public ConditionsBuilder falseLiteral()
+        {
+            return when("FALSE");
+        }
```

Comment:
nit: `trueLiteral` and `falseLiteral` unused...unless they're used in a later patch...